src/dvc_task/app/filesystem.py

```diff
@@ -122,6 +126,7 @@ def _iter_folder(
                     path = os.path.join(folder, filename)
                     try:
                         with open(path, "rb") as fobj:
+                            lock(fobj, LOCK_SH)
```

Comment:
does this need a matching `unlock()` call?

Reply:
> does this need a matching `unlock()` call?

> fobj.close() will automatically unlock it.